### ISSUE TITLE
Search result disappears unexpectedly when you amend the search

### ISSUE THREAD

**Issue by sarahneeson** [ServiceInnovationLab/FSD#201]
TL;DR
In the amended search below, why has the result for Tūranga Ararau dropped off, when their location is a street address in Gisborne?

---

To illustrate the problem, you need to conduct a particular search, then amend the search criteria and re-initiate the search.

Initial search:
Enter topic or organisation: "Turanga" and select Search button
https://serviceinnovationlab.github.io/FSD/#/FSD/?keyword=Turanga&radius=25

Results include:
1. [Turanga Social Services](https://serviceinnovationlab.github.io/FSD/#/FSD/service/6047)
2. [Tūranga Ararau](https://serviceinnovationlab.github.io/FSD/#/FSD/service/19369)
3. [ARC Counselling Services - Turangi](https://serviceinnovationlab.github.io/FSD/#/FSD/service/8343)

Amended search:
Start typing an address, select "Gisborne"
https://serviceinnovationlab.github.io/FSD/#/FSD/?keyword=Turanga&latitude=-38.66137&longitude=178.019055&radius=25&region=Gisborne

Results include:
1. [Turanga Social Services](https://serviceinnovationlab.github.io/FSD/#/FSD/service/6047)

Results **no longer include**:
[Tūranga Ararau](https://serviceinnovationlab.github.io/FSD/#/FSD/service/19369)
[ARC Counselling Services - Turangi](https://serviceinnovationlab.github.io/FSD/#/FSD/service/8343)

It makes sense that ARC Counselling Services would drop off, as they are based in Turangi.
However, why has Tūranga Ararau dropped off, when their location is a street address in Gisborne?

Screenshot of initial search:
![Initial search.PNG](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/5a2613c599d956001145c2b3/125516c66e82c728ace21e0d46c19a93378b82a5cffaba42b65ef8d665043761260ea62864c7b96aea412a595c0d46ff170b5a46f4af892ba0e46f70835459ecc3714d36976345f3e5db55ea43a82a994e82bcbd8f49c2e03525ef82874b391a1755.PNG)

Screenshot of amended search:
![Amended search.PNG](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/5a2613c599d956001145c2b3/125516c66e82c728ace21e0d46c9999f2d8686adcffaba42b65ef8d665043761260ea62864c7b96aea412a595c0d46ff170b5a46f4af892ba0e46f70835459ecc3714d36976345f3e5db55ea43a82a994e8cbcbd8f49c2e03524e58b8c4b3a161755.PNG)

**Comment by sarahneeson** [comment]
TL;DR
In the amended search below, why has the result for Tūranga Ararau dropped off, when their location is a street address in Gisborne?
-------------------------------------------------------------------------------------------
To illustrate the problem, you need to conduct a particular search, then amend the search criteria and re-initiate the search.

Initial search:
Enter topic or organisation: "Turanga" and select Search button
https://serviceinnovationlab.github.io/FSD/#/FSD/?keyword=Turanga&radius=25

Results include:
[Turanga Social Services](https://serviceinnovationlab.github.io/FSD/#/FSD/service/6047)
[Tūranga Ararau](https://serviceinnovationlab.github.io/FSD/#/FSD/service/19369)
[ARC Counselling Services - Turangi](https://serviceinnovationlab.github.io/FSD/#/FSD/service/8343)

Amended search:
Start typing an address, select "Gisborne"
https://serviceinnovationlab.github.io/FSD/#/FSD/?keyword=Turanga&latitude=-38.66137&longitude=178.019055&radius=25&region=Gisborne

Results no longer include:
[Tūranga Ararau](https://serviceinnovationlab.github.io/FSD/#/FSD/service/19369)
[ARC Counselling Services - Turangi](https://serviceinnovationlab.github.io/FSD/#/FSD/service/8343)

It makes sense that ARC Counselling Services would drop off, as they are based in Turangi.
However, why has Tūranga Ararau dropped off, when their location is a street address in Gisborne?

Screenshot of initial search:
![Initial search.PNG](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/5a2613c599d956001145c2b3/125516c66e82c728ace21e0d46c19a93378b82a5cffaba42b65ef8d665043761260ea62864c7b96aea412a595c0d46ff170b5a46f4af892ba0e46f70835459ecc3714d36976345f3e5db55ea43a82a994e82bcbd8f49c2e03525ef82874b391a1755.PNG)

Screenshot of amended search:
![Amended search.PNG](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/5a2613c599d956001145c2b3/125516c66e82c728ace21e0d46c9999f2d8686adcffaba42b65ef8d665043761260ea62864c7b96aea412a595c0d46ff170b5a46f4af892ba0e46f70835459ecc3714d36976345f3e5db55ea43a82a994e8cbcbd8f49c2e03524e58b8c4b3a161755.PNG)

**Comment by sarahneeson** [comment]
@choirboy07 Have I captured everything I need to in this ticket?

**Comment by lamorrison** [comment]
I've investigated this and found that the search result was being ignored because the address didn't look valid. In this case the address doesn't contain a street number (or a Postcode) and it was being interpreted as not a valid location.

When a location search (either for a region or a street address) is performed, there is a filter which removes results which can't be represented on the map. This will catch any results without latitude and longitude, but also filters out results where the address doesn't look right.

The existing code filters out addresses which don't contain a number (such as a street number or postcode). The address for Tūranga Ararau is 'Cnr Kahutia and Bright Streets, Gisborne', which was presented correctly in the keyword search but filtered out for the Gisborne region search.

We have made a change to the filtering code so that addresses will only be rejected if they are entirely blank. This will make search results for Tūranga Ararau behave as expected. 

It also raises the question of whether we should ever filter out results based on their address - if for some reason the address is missing but we do have latitude and longitude data it seems reasonable to show the result on the map.

Because the catalogue.data.govt.nz server is down today we haven't been able to finish this story but we're confident that we've found the cause of this issue and it doesn't need further investigation.

**Comment by choirboy07** [comment]
Absolutely agree that with the approach that where and address is missing or is not "valid" and the lat/Long is present use the lat/Long and do not filter out the result.

This fix for 201 explains part of the problems in 184 but not where all the addresses appear to be valid (Sarah's example) in the card.

**Comment by lamorrison** [comment]
Thanks Paul, we'll make the change you have agreed with to not worry about missing addresses.

It looks like the server is back up today so I'll try to get you some more information about the other search issues.

**Comment by CharlotteHinton** [comment]
@choirboy07 @sarahneeson ready for you to review

**Comment by sarahneeson** [comment]
Tested 28/3/19 at 2.35pm in Chrome, looks to be working!
![Turanga Ararau 1.PNG](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/5a2613c599d956001145c2b3/125516c66e82c728ace21e0d46dc8188228c84a8cfc8ad42b65ce5d478046b313916ac217586f76ae5002e025e574dfe490b0917f9fb817da6b63a22d95c02e5836c4d68d47f40f2ecdd42e64bef62994597b7bb804dc5ed3e22e48c8f4a3d1a051d9bb2.PNG)

**Comment by sarahneeson** [comment]
Also tried searching with a street address, and it shows up.  Yay :)
![Turanga Ararau 2.PNG](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/5a2613c599d956001145c2b3/125516c66e82c728ace21e0d46dc8188228c84a8cfc8ad42b65ce5d47b046b313916ac217586f76ae5002e025e574dfe490b0917f9fb817da6b63a22d95c02e5836c4d68d47f40f2ecdd42e64bef62994597b7bb804dc5ed3e22e48c8e47361f0c1f9bb2.PNG)

**Comment by sarahneeson** [comment]
@choirboy07 - note that I have dragged this ticket into Accepted
